src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -440,9 +462,67 @@ class Manager : public DoutPrefixProvider {
         if (ret < 0) {
           ldpp_dout(this, 1) << "ERROR: failed to remove entries and/or lock queue up to: " << end_marker <<  " from queue: " 
             << queue_name << ". error: " << ret << dendl;
+          return;
         } else {
-          ldpp_dout(this, 20) << "INFO: removed entries up to: " << end_marker <<  " from queue: " 
-          << queue_name << dendl;
+          ldpp_dout(this, 20) << "INFO: removed entries up to: " << end_marker <<  " from queue: " << queue_name << dendl;
+        }
+
+        // reserving and committing the migrating entries
+        if (migration_in_process) {
```

Comment:
you can just check that `entries_to_migrate.size() > 0` and the nyou don't need the `migration_in_process` flag

Reply:
done